modules/dxp/apps/saml/saml-persistence-service/src/main/java/com/liferay/saml/persistence/internal/upgrade/v3_0_0/SamlIdpSpSessionUpgradeProcess.java

```diff
@@ -38,35 +38,40 @@ protected void doUpgrade() throws Exception {
 					new AlterTableAddColumn("samlPeerBindingId", "LONG null"));
 			}
 
+			int samlIdpSpSessionId = 0;
 			int samlIdpSpSessionIdOffset = 0;
 
 			try (PreparedStatement preparedStatement =
-					connection.prepareStatement(
-						"select min(samlIdpSpSessionId) - 1 from " +
-							"SamlIdpSpSession");
-				ResultSet resultSet = preparedStatement.executeQuery()) {
+					 connection.prepareStatement(
+						 "select min(samlIdpSpSessionId) from SamlIdpSpSession");
+				 ResultSet resultSet = preparedStatement.executeQuery()) {
 
 				if (resultSet.next()) {
-					samlIdpSpSessionIdOffset = resultSet.getInt(1);
+					samlIdpSpSessionId = resultSet.getInt(1);
+
+					samlIdpSpSessionIdOffset = samlIdpSpSessionId - 1;
 				}
 			}
 
 			int latestSamlPeerBindingId = _getLatestSamlPeerBindingId();
 
+			int samlPeerBindingId =
+				samlIdpSpSessionId +
+				(-samlIdpSpSessionIdOffset + latestSamlPeerBindingId);
+
 			runSQL(
 				StringBundler.concat(
 					"insert into SamlPeerBinding (samlPeerBindingId, ",
 					"companyId, createDate, userId, userName, deleted, ",
 					"samlNameIdFormat, samlNameIdNameQualifier, ",
 					"samlNameIdSpProvidedId, samlNameIdValue, ",
-					"samlPeerEntityId) select min(samlIdpSpSessionId) + ",
-					-samlIdpSpSessionIdOffset + latestSamlPeerBindingId,
-					", companyId, min(createDate), userId, userName, '0' as ",
-					"deleted, nameIdFormat, null as nameIdNameQualifier, null ",
-					"as nameIdSpProvidedId, nameIdValue, samlSpEntityId from ",
-					"SamlIdpSpSession group by companyId, userId, userName, ",
-					"samlSpEntityId, nameIdFormat, nameIdValue, ",
-					"samlSpEntityId"));
+					"samlPeerEntityId) select ", samlPeerBindingId,
```

Comment:
We can't use the same SamlPeerBindingId to instead multiple rows. The original code included `min(samlIdpSpSessionId)` which gets the lowest ID for each record created by the "group by".
If we need to do this in Java then we need to do this in a loop.

Reply:
Understood, you're right, I'm reverting this as we discussed on Slack